zirgen/dsl/test/binary_adder.zir

```diff
@@ -0,0 +1,63 @@
+// RUN: zirgen --test %s 2>&1 | FileCheck %s
+
+component BitReg(x: Val) {
+  r := Reg(x);
+  r * (r - 1) = 0;
+  r
+}
+
+component Not(x: BitReg) {
+  BitReg(1 - x)
+}
+
+component And(x: BitReg, y: BitReg) {
+  BitReg(x * y)
+}
+
+component Or(x: BitReg, y: BitReg) {
+  BitReg(x + y - x * y)
+}
+
+component Xor(x: BitReg, y: BitReg) {
+  BitReg(x + y - 2 * x * y)
+}
+
+component Xor3(x: BitReg, y: BitReg, z: BitReg) {
+  Xor(Xor(x, y), z)
+}
+
+component HalfAdder(x: BitReg, y: BitReg) {
+  sum := Xor(x, y);
+  carry := And(x, y);
+}
+
+component FullAdder(x: BitReg, y: BitReg, c: BitReg) {
+  sum := Xor(Xor(x, y), c);
```

Comment:
seems like either this line should use Xor3 defined above, or the otherwise-unused Xor3 component ought to be dropped from the example